### PR TITLE
Improve docker setup

### PR DESCRIPTION
**Pull request by ChristophB**
This pull request introduces the production-ready WSGI server waitress and some improvements to the Dockerfile. The Docker build succeeds and the web server starts, but I was not able to test the actual functionality of the graph pipeline.

There is also now a `docker-compose.yml` to make it easier to build the image and deploy a container.

I find it a bit strange that the spacy model de_dep_news_trf contributes ~10GB to the image size, but the docs mention a size of ~400MB.

### CODE REVIEW

**Comment by ChristophB** [comment]
Added a docker build and publish workflow. Feel free to drop the latest commit.

File: Dockerfile [hunk 1]
```diff
@@ -48,43 +47,18 @@ RUN apt-get update --yes && \
...
-    ${PYTHON} -m pip install spacy spacy-lookups-data spacy-transformers && \
-    ${PYTHON} -m pip install torch==1.11.0+cpu torchvision==0.12.0+cpu torchaudio==0.11.0+cpu -f https://download.pytorch.org/whl/cpu/torch_stable.html && \
-    ${PYTHON} -m pip install altair networkx pyvis umap-learn[plot] && \
-    ${PYTHON} -m pip install python-Levenshtein fuzzywuzzy && \
-    ${PYTHON} -m pip install bratiaa && \
-    ${PYTHON} -m pip install flask && \
-    ${PYTHON} -m pip install pyyaml && \
-    ${PYTHON} -m pip install scikit-network && \
-    ${PYTHON} -m pip install --upgrade requests && \
-    ${PYTHON} -m spacy download de_core_news_sm && \
```
Comment (ChristophB) [inline Dockerfile:83]: Not sure if this dependency can be dropped.

**Comment by fmatthies** [comment]
Thanks for the changes. Where did you check the size of the `de_dep_news_trf ` model? It's really just ~400MB; at least on my local machine.

**Review by fmatthies — APPROVED** [review]
accepted

**Comment by ChristophB** [comment]
> Thanks for the changes. Where did you check the size of the `de_dep_news_trf ` model? It's really just ~400MB; at least on my local machine.

I simply built the image with and without downloading the `de_dep_news_trf` model and compared the size of both images.

It looks like downloading `de_dep_news_trf` will (re)install PyTorch and other packages as well.

**Comment by fmatthies** [comment]
Hm, so it would be best to deduce the dependencies of the model and don't install them beforehand, because `spacy` probably enforces versions as well and therefore reinstalls them if necessary.

**Comment by ChristophB** [comment]
With [6d8f7db](https://github.com/Onto-Med/concept-graphs/pull/2/commits/6d8f7db867d279dfbe57531a086b818347c8d253) the image now has a size of 4.86GB.

**Comment by fmatthies** [comment]
sounds good. how did you go about determining the fixed versions?
PS: I have another branch reworks the negation detection -> so you need not make any further changes to `negex` or `benepar` related stuff :)

**Comment by ChristophB** [comment]
> how did you go about determining the fixed versions?

I built the container and executed `pip freeze` inside.

**Pull request merged**